llvm/test/CodeGen/EraVM/indexed-memops-gep-dominate-bug.ll

```diff
@@ -0,0 +1,73 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 2
+; RUN: llc -O3 < %s | FileCheck %s
```

Comment:
An IR level test that only runs EraVMIndexedMemOpsPrepare could be a better option here.

Reply:
I agree, but we also need to run verify pass, since there was an assert. Because of this, I have added support for new pass manager to `EraVMIndexedMemOpsPrepare` similar to [this](https://github.com/matter-labs/era-compiler-llvm/commit/5b885a6ece97b8df7cb8bdedda11e50e0cf00723) one. Unfortunately there is a catch, `EraVMIndexedMemOpsPrepare` is a loop pass and LLVM new pass manager machinery is adding `LoopSimplifyPass` and `LCSSAPass` before loop passes. This is a problem since if we want to run `EraVMIndexedMemOpsPrepare` with `opt` we would get different pipeline than we use. Because of this, I dropped new pass manager support for `EraVMIndexedMemOpsPrepare`.
Please note that I changed this test to stop after verify pass, so we make sure assert is not triggered.